test-suite/src/data_type/int128.rs

```diff
@@ -0,0 +1,106 @@
+use {
+    crate::*,
+    gluesql_core::{data::ValueError, prelude::DataType, prelude::Value::*},
+};
+
+test_case!(int128, async move {
+    run!(
+        "CREATE TABLE Item (
+        field_one INT(128),
+        field_two INT(128),
+    );"
+    );
+    run!("INSERT INTO Item VALUES (1, -1), (-2, 2), (3, 3), (-4, -4);");
+
+    let parse_i128 = |text: &str| -> i128 { text.parse().unwrap() };
+
+    let max_str = "170141183460469231731687303715884105728";
+    let min_str = "-170141183460469231731687303715884105729";
+
+    test!(
+        Err(ValueError::FailedToParseNumber.into()),
+        &format!("INSERT INTO Item VALUES ({}, {})", max_str, max_str)
+    );
+
+    // cast i128::MAX+1
+    test!(
+        Err(ValueError::LiteralCastToDataTypeFailed(DataType::Int128, max_str.to_string()).into()),
+        &format!("select cast({} as INT(128)) from Item", max_str)
+    );
+
+    // cast i128::MIN-1
+    test!(
+        Err(ValueError::LiteralCastToDataTypeFailed(DataType::Int128, min_str.to_string()).into()),
+        &format!("select cast({} as INT(128)) from Item", min_str)
+    );
+
+    // lets try some valid SQL
+    test!(
+        Ok(select!(
+            field_one        | field_two
+            I128               |    I128;
+            1                   parse_i128("-1");
+            parse_i128("-2")        2;
+            3                      3;
+            parse_i128("-4")      parse_i128("-4")
+        )),
+        "SELECT field_one, field_two FROM Item"
+    );
```

Comment:
This is very nit request.
Could you make this looks like table shape?
e.g.
```rust
test!(
    Ok(select!(
        field_one        | field_two
        I128             | I128;
        1                  parse_i128("-1");
        parse_i128("-2")   2;
        3                  3;
        parse_i128("-4")   parse_i128("-4")
    )),
    "SELECT field_one, field_two FROM Item"
);
```

btw. thanks to use `parse_i128`.
`select!` macro needs to be upgrade to handle `-10` like negative numbers but.. current it does not.